### PR TITLE
Documentation: Update main.js exports migration

### PR DESCRIPTION
**Pull request by yannbf**
Added some notes about types in js files

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -335,6 +337,17 @@ const config: StorybookConfig = {
 export default config;
 ```
 
+In Javascript projects using `main.js`, it's also possible to use the `StorybookConfig` type (for autocompletion, not type safety), via the JSDoc @type tag:
+
+```ts
+/** @type { import('@storybook/react-vite').StorybookConfig } */
+const config: StorybookConfig = {
```
Comment (shilman) [inline MIGRATION.md:344]:
```suggestion
const config = {
```

**Review by shilman — CHANGES_REQUESTED** [review]
LGTM to me except for what I think is a typo

File: MIGRATION.md [hunk 2]
```diff
@@ -335,6 +337,17 @@ const config: StorybookConfig = {
 export default config;
 ```
 
+In Javascript projects using `main.js`, it's also possible to use the `StorybookConfig` type (for autocompletion, not type safety), via the JSDoc @type tag:
```
Comment (JReinhold) [inline MIGRATION.md:340]:
```suggestion
In JavaScript projects using `main.js`, it's also possible to use the `StorybookConfig` type (for autocompletion, not type safety), via the JSDoc @type tag:
```

**Review by JReinhold — APPROVED** [review]

File: MIGRATION.md (hunk 1 above)
Comment (yannbf) [reply MIGRATION.md]: Great catch!!

File: MIGRATION.md (hunk 2 above)
Comment (yannbf) [reply MIGRATION.md]: thanks!

**Review by shilman — APPROVED** [review]

**Pull request merged**